### PR TITLE
Wonhee updates the workgroup, best practices in R, and running GPU pages 

### PR DESCRIPTION
**Pull request by wolee101**
Wonhee edited the following 3 pages to update content, enhance formatting and organization, and align with style guidelines. Major changes are noted in parentheses:

- /_policies/workgroups.md (Consolidated redundant info and reorganized content for clarity)
- /_user_guide/best_practices_r.md (Tested code and added more info including a new image)
- /blog/posts/2024-03-28-run-gpu.md (Tested code and rewrote the Colab section with examples, code, and outputs. Note that this page uses the same Keras example as the Yen GPU page, but applied to Sherlock and Colab.)

added the following 1 image file:

- /assets/images/yahtzee_histogram.png

and deleted the following 3 files that are in the old " _blog" folder.

- docs/_blog/2020-03-04-text_processing_word_embeddings.md 
- docs/_blog/2021-01-20-performance_tuning.md 
- docs/_blog/2023-09-18-download_files_wget.mdand

---------------- (after PR reviews)

Renamed file from docs/blog/posts/2024-03-28-run-gpu.md to docs/blog/posts/2024-03-28-colab_gpu.md and removed the redundant Shrlock section from the page and formatted the Colab section accordingly. 

Added the author info for wolee to the docs/blog/.authors.yml file.

### CODE REVIEW

File: docs/_policies/workgroups.md [hunk 1]
```diff
@@ -1,29 +1,22 @@
...
 
-## Adding/Removing Users to your Workgroup
-If you are an Administrator of a workgroup, you can use the Workgroup Manager interface to easily add users to and remove users from that workgroup. Adding users to a workgroup immediately grants them permissions to whatever resource that workgroup controls access to. Likewise, removing users revokes those permissions. You might want to do this if you no longer want someone to have access to your project space or as part of regular auditing of external collaborators you have sponsored access to the Yens for.
+## What is a Workgroup?
 
-Users can be added as Members and/or Administrators of a workgroup. While Members are simply granted permission to the resource controlled by the workgroup, Administrators have the power to add *other* users as Members/Administrators to the workgroup.
+The [Stanford Workgroup Manager](https://workgroup.stanford.edu/){:target="_blank"}, is a web interface that allows Stanford community members to define groups of other community members based on their SUNet IDs for use in various applications. Each workgroup is assigned a unique name that succintly describes its purpose.
```
Comment (alexstorer) [inline docs/_policies/workgroups.md:5]: Remove `,`

Comment (alexstorer) [reply docs/_policies/workgroups.md:5]: Maybe reword this to not lead with the workgroup manager.  Also -- spelling error with "succinctly", and stray comma.

> A *workgroup* is a list of members in a group, identified by their SUNet IDs, and given a name that uniquely identifies it, for use on the Yen servers and at Stanford more broadly. The [Stanford Workgroup Manager](https://workgroup.stanford.edu/){:target="_blank"} is a web interface that allows Stanford community members to manage workgroup membership.

File: docs/_policies/workgroups.md [hunk 2]
```diff
@@ -1,29 +1,22 @@
...
+## What is a Workgroup?
 
-Users can be added as Members and/or Administrators of a workgroup. While Members are simply granted permission to the resource controlled by the workgroup, Administrators have the power to add *other* users as Members/Administrators to the workgroup.
+The [Stanford Workgroup Manager](https://workgroup.stanford.edu/){:target="_blank"}, is a web interface that allows Stanford community members to define groups of other community members based on their SUNet IDs for use in various applications. Each workgroup is assigned a unique name that succintly describes its purpose. 
 
-!!! warning
-    Given that Administrators of a workgroup have the power to add other users, please be mindful of who you grant Administrator privileges to. It is *your responsibility* to audit both the Administrators and Members of your workgroup.
+Relevant to the Yens, the Workgroup Manager helps control access to individual [project space](/_policies/services){:target="_blank"} on ZFS (e.g., gsb-rc:[faculty-SUNet]-[projectname]). For example, if your project is stored at `/zfs/projects/faculty/jdoe-housing`, there will be a corresponding workgroup named `gsb-rc:jdoe-housing`.
```
Comment (alexstorer) [inline docs/_policies/workgroups.md:7]: I think we can reword this to be more clear?

Workgroups on the Yens are used to manage system logins as well as access to [project space](/_policies/services){:target="_blank"}. Each project space has a corresponding workgroup, so members of the workgroup `gsb-rc:jdoe-housing` have read/write access to data stored at `/zfs/projects/faculty/jdoe-housing`. Members of these project workgroups will also have login access to the Yen servers more generally.

File: docs/_policies/workgroups.md [hunk 3]
```diff
@@ -1,29 +1,22 @@
...
 
+## How to Add or Remove Users to Your Workgroup?
 
-For instructions on how to perform these operations, use this [helpful page](https://uit.stanford.edu/service/workgroup/add-remove-members) created by Stanford UIT that illustrates these processes with screenshots.
+If you are an Administrator of a workgroup, you can use the Stanford Workgroup Manager to easily add users to and remove users from that workgroup. Adding users to a workgroup immediately grants them permissions to whatever resource that workgroup controls access to. Likewise, removing users revokes those permissions. You might want to do this if you no longer want someone to have access to your project space or as part of regular auditing of external collaborators you have sponsored access to the Yens for.
```
Comment (alexstorer) [inline docs/_policies/workgroups.md:11]: It's not "immediate" -- let's remove that word and add a warning that workgroup changes take up to 24 hours to propagate to the Yen servers.

File: docs/_policies/workgroups.md [hunk 4]
```diff
@@ -1,29 +1,22 @@
...
+!!! Warning
+    Administrators of a workgroup have the power to add **other** users. Please be cautious about whom you grant Administrator privileges to, as it is **your responsibility** to audit both the Administrators and Members of your workgroup.
 
-Your workgroup should have a name that is clear from the name of your project space.  For example, if your project is stored at `/zfs/projects/faculty/jdoe-housing`, there will be a workgroup called `gsb-rc:jdoe-housing`.
+## How to Add a Collaborator to Your Yen Project Space?
 
-Faculty and students should use the [workgroup manager](https://workgroup.stanford.edu) to add members, and optionally, administrators to their projects.  This is the same system used for [yen access](/_policies/collaborators).  For step-by-step instructions on how to use the workgroup manager, see our [workgroup guide](/yen/workgroups.html).
+Adding (or removing) members to your project space is a self-service task for project administrators. To add a collaborator, simply add them to the workgroup associated with the project. More details can be found here [here](/_policies/collaborators){:target="_blank"}.
```
Comment (alexstorer) [inline docs/_policies/workgroups.md:22]: Let's add some language about finding the workgroup if they're lost (or asking if they need help)?

A clarification?
> More details about adding non-GSB collaborators can be found here [here](/_policies/collaborators){:target="_blank"}

File: docs/_user_guide/best_practices_r.md [hunk 5]
```diff
@@ -1,20 +1,28 @@
...
 😱 cluster_fork <- makeForkCluster(detectCores()) 😱
 ```
 
-Don't use `detectCores`.  This will ask the machine how many cores there are, and then your code will try to use them all - **don't do this on a shared environment**.  Instead, replace `detectCores()` with 4, or some other reasonably scaled amount of resources.  It's often a good idea to benchmark your code first before trying to scale up - consider [this guide](https://jstaf.github.io/hpc-r/parallel/) or [this one](https://bookdown.org/rdpeng/rprogdatascience/parallel-computation.html) to understand how your code might benefit from parallelization.
+Don't use `detectCores`. It checks the machine for the total number of available cores and attempts to use all of them. Instead, replace `detectCores()` with a fixed number, such as 4, or another reasonably scaled allocation of resources. It's also a good practice to benchmark your code first before scaling up. Consider [this guide](https://jstaf.github.io/hpc-r/parallel/){:target="_blank"} or [this one](https://bookdown.org/rdpeng/rprogdatascience/parallel-computation.html){:target="_blank"} to understand how your code might benefit from parallelization.
+
+!!! Warning
```
Comment (alexstorer) [inline docs/_user_guide/best_practices_r.md:13]: This warning feels redundant?

File: docs/_user_guide/best_practices_r.md [hunk 6]
```diff
@@ -1,20 +1,28 @@
...
+## Parallelization Under the Hood
 
-## An instructive example
+Even if you’re not explicitly running parallel code, some packages may use parallelization "under the hood." If you’re unsure, learn how to [monitor usage](/_user_guide/best_practices_monitor_usage/){:target="_blank"}. If your code runs in parallel automatically, figure out which options to pass to the functions you’re using in R to limit the number of cores. For example, in the [ranger](https://rdocumentation.org/packages/ranger/versions/0.15.1) package, the `num.threads` option defaults to the total number of cores (!!!). However, you can—and should-override this manually to prevent overuse of resources.
```
Comment (alexstorer) [inline docs/_user_guide/best_practices_r.md:18]: Let's change `!!!` to `😱` to match the above?

Comment (alexstorer) [reply docs/_user_guide/best_practices_r.md:18]: The two dashes with `can-and should-override` render differently, is that on purpose? If we want the big ones, let's explicitly use the "em dash" `—` to avoid this?

> you can—and should—override

File: docs/_user_guide/best_practices_r.md [hunk 7]
```diff
@@ -31,13 +39,22 @@ getRolls <- function(x) {
...
+
+# Perform the parallel computation
 res <- mclapply(1:100,getRolls,mc.cores=nc)
+
+# Create a histogram
 hist(unlist(res))
 ```
 
-This uses 4 cores on my local machine to apply this function.
+This uses 4 cores on your local machine to apply this function. If you display or save the results as a histogrm (e.g., `yahtzee_histogram.png`), it will look similar to this:
```
Comment (alexstorer) [inline docs/_user_guide/best_practices_r.md:58]: I don't think the histogram adds much here, particularly since it doesn't show anything about parallelism.

Also, histogrm is misspelled :)

File: docs/_user_guide/best_practices_r.md [hunk 8]
```diff
@@ -1,20 +1,28 @@
...
+## Parallelization Under the Hood
 
-## An instructive example
+Even if you’re not explicitly running parallel code, some packages may use parallelization "under the hood." If you’re unsure, learn how to [monitor usage](/_user_guide/best_practices_monitor_usage/){:target="_blank"}. If your code runs in parallel automatically, figure out which options to pass to the functions you’re using in R to limit the number of cores. For example, in the [ranger](https://rdocumentation.org/packages/ranger/versions/0.15.1) package, the `num.threads` option defaults to the total number of cores (!!!). However, you can—and should-override this manually to prevent overuse of resources.
 
-Let's consider the following function, which considers how many rolls of 5 dice it takes to get a "Yahtzee" (all of a kind):
+## An Example of Parallelized Simulation
 
-```R
+Let's consider the following function, which determins how many rolls of 5 dice are needed to achieve a "Yahtzee" (all dice showing the same value):
```
Comment (alexstorer) [inline docs/_user_guide/best_practices_r.md:22]: typo on `determins`

**Review by alexstorer — CHANGES_REQUESTED** [review]

File: docs/blog/posts/2024-03-28-run-gpu.md [hunk 9]
```diff
@@ -47,66 +40,76 @@ chmod +x Miniconda3-latest-Linux-x86_64.sh
...
 ```
 Source bash profile to apply changes:
 
-```bash
+```title="Terminal Command"
 source ~/.bash_profile
 ```
-
-#### Create conda env
+### Create Conda Environment
```
Comment (alexstorer) [inline docs/blog/posts/2024-03-28-run-gpu.md:67]: We recommend against using conda pretty much everywhere, I think the default sherlock packages might be able to do this out of the box?  That's what I verified when I wrote the Sherlock page: http://127.0.0.1:8000/_user_guide/sherlock/#deep-learning-on-gpu

**Review by alexstorer — CHANGES_REQUESTED** [review]

File: docs/_user_guide/best_practices_r.md (hunk 6 above)
Comment (wolee101) [reply docs/_user_guide/best_practices_r.md:18]: I intended to use an em dash in VS Code but didn’t realize it doesn’t render correctly. I changed it to "you can override."

File: docs/_user_guide/best_practices_r.md (hunk 7 above)
Comment (wolee101) [reply docs/_user_guide/best_practices_r.md:58]: Deleted

File: docs/blog/posts/2024-03-28-run-gpu.md (hunk 9 above)
Comment (wolee101) [reply docs/blog/posts/2024-03-28-run-gpu.md:67]: I thought about that too. I guess the Sherlock page I referenced was not authored by you. Since it still mentioned the conda environment, I included it in the this page. Since the content is largely the same, should we just remove this blog page? My primary contribution to this page was rewriting the Colab section, but I'm not sure how helpful it is. If it is useful, we cam create a separate page for Colab. Thoughts?

File: docs/_policies/workgroups.md (hunk 4 above)
Comment (wolee101) [reply docs/_policies/workgroups.md:22]: Did you mean adding language related to finding a workgroup in the Stanford Workgroup Manager, or something else? If it's the former, this link (https://uit.stanford.edu/service/workgroup/add-remove-members) is already included on the page. If not, do you have any suggestions?

File: docs/blog/posts/2024-03-28-colab_gpu.md [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+422/422 ━━━━━━━━━━━━━━━━━━━━ 82s 107ms/step - accuracy: 0.9884 - loss: 0.0345 - val_accuracy: 0.9928 - val_loss: 0.0282
+Epoch 14/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 82s 106ms/step - accuracy: 0.9896 - loss: 0.0321 - val_accuracy: 0.9920 - val_loss: 0.0285
+Epoch 15/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 82s 107ms/step - accuracy: 0.9882 - loss: 0.0360 - val_accuracy: 0.9923 - val_loss: 0.0275
+Test loss: 0.023274529725313187
+Test accuracy: 0.9919000267982483
+```
+
+Enusre that the **paths** for loading data and saving outputs and are correctly set as needed. Below is an example of how to save results, the trained model, and predictions to Google Drive:
```
Comment (alexstorer) [inline docs/blog/posts/2024-03-28-colab_gpu.md:197]: Typo

File: docs/blog/posts/2024-03-28-colab_gpu.md [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+Running the above code will generate files in your Google Drive, with output messages like this:
+
+```{ .yaml .no-copy title="colab_mnist.ipynb" }
+Evaluation results saved to ./outputs/results.txt
+Predictions saved to ./outputs/predictions.npy
+Model saved to ./outputs/mnist_model.keras
+313/313 ━━━━━━━━━━━━━━━━━━━━ 4s 14ms/step
+```
+
+!!! Tip
```
Comment (alexstorer) [inline docs/blog/posts/2024-03-28-colab_gpu.md:235]: This does not render correctly

**Review by alexstorer — CHANGES_REQUESTED** [review]

File: docs/blog/posts/2024-03-28-colab_gpu.md [hunk 12]
```diff
@@ -0,0 +1,236 @@
+---
+date:
```
Comment (alexstorer) [inline docs/blog/posts/2024-03-28-colab_gpu.md:2]: Make sure you add yourself as an author!
https://squidfunk.github.io/mkdocs-material/tutorials/blogs/navigation/#defining-authors

```
  authors:
      - jeffotter
```

**Review by alexstorer — CHANGES_REQUESTED** [review]

File: docs/blog/posts/2024-03-28-run-gpu.md (hunk 9 above)
Comment (wolee101) [reply docs/blog/posts/2024-03-28-run-gpu.md]: Let me know if the Colab page looks good or need any edits.

File: docs/blog/posts/2024-03-28-colab_gpu.md (hunk 12 above)
Comment (wolee101) [reply docs/blog/posts/2024-03-28-colab_gpu.md:2]: Done!

File: docs/blog/posts/2024-03-28-colab_gpu.md (hunk 11 above)
Comment (wolee101) [reply docs/blog/posts/2024-03-28-colab_gpu.md:238]: It looks fine to me. Let me know if it still doesn't render correctly for you.

File: docs/_policies/workgroups.md (hunk 1 above)
Comment (alexstorer) [reply docs/_policies/workgroups.md:5]: `succintly` is still misspelled

**Review by alexstorer — CHANGES_REQUESTED** [review]

**Review by alexstorer — APPROVED** [review]
